compiler/rustc_trait_selection/src/traits/misc.rs

```diff
@@ -49,7 +50,17 @@ pub fn can_type_implement_copy<'tcx>(
                     continue;
                 }
                 let span = tcx.def_span(field.did);
-                let cause = ObligationCause::dummy_with_span(span);
+                // If the ADT has substs, point to the cause we are given.
+                // If it does not, then this field probably doesn't normalize
+                // to begin with, and point to the bad field.
+                let cause = if field
+                    .ty(tcx, traits::InternalSubsts::identity_for_item(tcx, adt.did))
+                    .has_param_types_or_consts()
+                {
+                    cause.clone()
+                } else {
+                    ObligationCause::dummy_with_span(span)
+                };
```

Comment:
This is...weird. If you don't do this, what changes?

Reply:
Some test starts pointing at a derive(Copy) saying that the type doesn't implement a trait, instead of the struct field.

Not at my computer, but it's some test that contains `<u32 as Iterator>::Item` in a struct that derives Copy.

Reply:
I've split out this hack into its own commit -- see aa4d68a for the regression in spans in issue-50480 (i.e. without this code).

There's probably something I don't understand about the ordering of the `impl Copy` correctness check and whatever check (wf?) that raises an error that the struct field `<i32 as Iterator>::Item` is invalid. Perhaps I can fix this in a better way than I am doing right now.

Reply:
Ugh, I see that this does fix a `Span`, but it still feels wrong to me. Kind of signals that this isn't the best diff. Anyways, can you just add a FIXME.